pkg/storage/secret/keeper/keeper.go

```diff
@@ -0,0 +1,34 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/storage/secret/encryption"
+	"github.com/grafana/grafana/pkg/storage/secret/vaultstorage"
+)
+
+// SecretKeeperService encapsulates two things:
+// 1- A service that encrypts and decrypts data.
+// 2- A service that stores and retrieves data, but this data is an encrypted value, which does not need to be encrypted in (1).
+type SecretKeeperService interface {
+	Encryption() encryption.EncryptionService
```

Comment:
This is possibly not needed in the interface itself - it could be wired in to the specific Keepers that need it instead. Can keep it here for now though!

Reply:
sqlKeeper := SqlKeeper(envelopeEncryptionSvc, kmsEncryptionService)
sqlKeeper.Store("hunter2")
...
within sqlKeeper.Store...
s.encService.Encrypt

======

awsKeeper := AwsKeeper()
awsKeeper.Store("hunter2")
...
within awsKeeper.Store...
s.client.Store("hunter2", kmsKeyId)